x-pack/plugins/endpoint/server/routes/alerts/lib/index.ts

```diff
@@ -3,28 +3,97 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { RisonValue, decode, encode } from 'rison-node';
 import { SearchResponse } from 'elasticsearch';
-import { IScopedClusterClient } from 'kibana/server';
+import { stringify } from 'query-string';
+import { IScopedClusterClient, KibanaRequest } from 'kibana/server';
 import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
-import { esKuery, esQuery } from '../../../../../../../src/plugins/data/server';
+import { esKuery, esQuery, Filter, TimeRange } from '../../../../../../../src/plugins/data/server';
 import { AlertEvent, Direction, EndpointAppConstants } from '../../../../common/types';
+import { EndpointAppContext } from '../../../types';
 import {
-  AlertSearchQuery,
+  AlertListRequestQuery,
+  AlertSearchParams,
   AlertSearchRequest,
   AlertSearchRequestWrapper,
   AlertSort,
+  UndefinedResultPosition,
 } from '../types';
-
 export { Pagination } from './pagination';
 
-function reverseSortDirection(order: Direction): Direction {
-  if (order === Direction.asc) {
-    return Direction.desc;
+/**
+ * Translates API search, sort, and pagination parameters to ES-ready parameters.
+ */
+export const getAlertSearchParams = async (
+  request: KibanaRequest<unknown, AlertListRequestQuery, unknown>,
+  endpointAppContext: EndpointAppContext
+): Promise<AlertSearchParams> => {
+  const config = await endpointAppContext.config();
+  const reqData: AlertSearchParams = {
+    // Filtering
+    query: request.query.query,
+    filters:
+      request.query.filters !== undefined
+        ? ((decode(request.query.filters) as unknown) as Filter[])
+        : ([] as Filter[]),
+    dateRange: ((request.query.date_range !== undefined
+      ? decode(request.query.date_range)
+      : config.alertResultListDefaultDateRange) as unknown) as TimeRange,
+
+    // Sorting
+    sort: request.query.sort,
+    order: request.query.order,
+
+    // Paging - Simple
+    pageIndex: request.query.page_index,
+    pageSize: request.query.page_size,
+
+    // Paging - Cursor-Based
+    searchAfter: request.query.after,
+    searchBefore: request.query.before,
```

Comment:
this might be simpler if we combine `searchAfter` and `searchBefore` into `cursor` and add another bool param `pageBackwards` to control the direction which would default to `false`